Добавлены тесты на класс Бургер.

elizavetamakarovavn-netizen opened the pull request:
При написании тестов использовались Mockito, JUnit 4. Покрытие класса Burger 100%, проверено с помощью JaCoCo

elizavetamakarovavn-netizen committed:
1.Добавлены тесты на класс Бургер

elizavetamakarovavn-netizen committed:
Добавлена оценка покрытия JaCoCo

okryukova commented on src/test/java/praktikum/BurgerAddIngredientTest.java:
```diff
@@ -0,0 +1,20 @@
+
+        burger.addIngredient(ingredient);
+
+        assertEquals(1, burger.ingredients.size());
```
⛔️Нужно исправить. Для юнит-тестов придерживаемся подхода: один тест, значит одна проверка. Если очень хочется несколько проверок -- тогда используем softAssertions. Поправь, пожалуйста, во всем коде

okryukova replied:
Не исправлено

okryukova commented on src/test/java/praktikum/BurgerGetPriceParameterizedTest.java:
```diff
@@ -0,0 +1,54 @@
+        this.expected = expected;
+    }
+
+    @Parameterized.Parameters
```
⚠️Можно улучшить. В параметризованных тестах для аннотации Parameterized.Parameters лучше использовать аргумент name: @Parameterized.Parameters(name = "Тестовые данные: {0} {1}"), где {0}, {1} - индексы параметров. Это повысит информативность проверки

okryukova commented on src/test/java/praktikum/BurgerSetBunsTest.java:
```diff
@@ -0,0 +1,20 @@
+import static org.mockito.Mockito.*;
+
+
+public class BurgerSetBunsTest {
```
⚠️Можно улучшить.  Не параметризированные тесты можно объединить в один класс

okryukova commented on src/test/java/praktikum/BurgerMoveIngredientTest.java:
```diff
@@ -0,0 +1,29 @@
+        burger.moveIngredient(0, 2);
+
+        assertTrue(burger.ingredients.get(0) == second);
+        assertTrue(burger.ingredients.get(1) == third);
```
⛔️Нужно исправить. Для юнит-тестов придерживаемся подхода: один тест, значит одна проверка. Если очень хочется несколько проверок -- тогда используем softAssertions. Поправь, пожалуйста, во всем коде

elizavetamakarovavn-netizen committed:
1. Тесты переписаны в соответствии с "один тест - одна проверка"
2. В параметризованные тесты к аннотации @Parameterized.Parameters добавлен аргумент name
